internal/decryption/output.go

```diff
@@ -21,39 +21,87 @@ import (
 	"go.mongodb.org/mongo-driver/bson"
 )
 
-type AuditLogOutput struct {
-	Warningf  func(lineNb int, format string, a ...interface{}) error
-	Error     func(lineNb int, err error) error
-	Errorf    func(lineNb int, format string, a ...interface{}) error
-	LogRecord func(lineNb int, logRecord interface{}) error
-}
-
-func buildOutput(out io.Writer) AuditLogOutput {
-	newLine := []byte{'\n'}
-	writeLine := func(value []byte) error {
-		if _, err := out.Write(value); err != nil {
-			return err
-		}
-		_, err := out.Write(newLine)
+type AuditLogOutput interface {
+	Warningf(lineNb int, format string, a ...interface{}) error
+	Error(lineNb int, err error) error
+	Errorf(lineNb int, format string, a ...interface{}) error
+	LogRecord(lineNb int, logRecord interface{}) error
+}
+
+type auditLogOutputImpl struct {
```

Comment:
[nit] not very go idiomatic to have `Impl`, prefer to give them names related to what they do than by what they are, `AuditLogOutput` could be `AuditLogPrinter`  and `auditLogOutputImpl` an `auditLogIOWriter`(?)

Also do we really need this interface, prefer interfaces as inputs for functions and concretes as outputs and also why is a good idea to define the interface close to the function that takes the expected contract